tensorflow_serving/util/net_http/server/internal/evhttp_server.cc

```diff
@@ -105,6 +105,9 @@ bool EvHTTPServer::Initialize() {
     return false;
   }
 
+  evhttp_set_allowed_methods(ev_http_,
+                             EVHTTP_REQ_GET | EVHTTP_REQ_POST | EVHTTP_REQ_HEAD |
+                             EVHTTP_REQ_PUT | EVHTTP_REQ_DELETE | EVHTTP_REQ_OPTIONS);
```

Comment:
which ones are not allowed by default? could you document them here?

Reply:
I read this from the documentation: (http://www.wangafu.net/~nickm/libevent-2.0/doxygen/html/http_8h.html#ae63a0370f59570e00753f7fb512a7f59).
Correct me if the actual behavior is different from that.

So the default allowed methods are GET, POST, HEAD, PUT, DELETE, which correspond to EVHTTP_REQ_GET, EVHTTP_REQ_POST, EVHTTP_REQ_HEAD, EVHTTP_REQ_PUT, EVHTTP_REQ_DELETE.
I added an extra one here, which is EVHTTP_REQ_OPTIONS for the OPTIONS request.
Without adding that flag, if I send OPTIONS request, the server will return 405.

Reply:
I feel like I should add more comments to my code?

Reply:
Right. Could you document that libevents doesn't allow OPTIONS by default?

Reply:
sure